source/slang/slang-ir-autodiff-cfg-norm.cpp

```diff
@@ -473,14 +473,18 @@ struct CFGNormalizationPass
                     loopEndPoint = falseEndPoint;
                     isLoopOnTrueSide = false;
                 }
+
```

Comment:
I am little worried about the assumptions being made here regarding the structures of the loop body here. If a user writes something like `for(;;) {}`, the loop body can be in any form so we won't even be able to map blocks into condition blocks or continue blocks. I wonder if the code is handling thoses cases.

Reply:
I'm working on removing the assumption of having a condition block (by inserting a dummy condition like `if(true)`).

After that the only requirement should be that the first block after block with the loop terminator is part of the loop itself